sdk/eventhub/azure-eventhub/azure/eventhub/_connection_string_parser.py

```diff
@@ -11,16 +12,26 @@ class EventHubConnectionStringProperties(DictMixin):
     Properties of a connection string.
     """
 
-    def __init__(self, **kwargs):
-        self._fully_qualified_namespace = kwargs.pop("fully_qualified_namespace", None)
-        self._endpoint = kwargs.pop("endpoint", None)
-        self._eventhub_name = kwargs.pop("eventhub_name", None)
-        self._shared_access_signature = kwargs.pop("shared_access_signature", None)
-        self._shared_access_key_name = kwargs.pop("shared_access_key_name", None)
-        self._shared_access_key = kwargs.pop("shared_access_key", None)
+    def __init__( # pylint: disable=unused-argument
+        self,
+        *,
+        fully_qualified_namespace: str,
+        endpoint: str,
+        eventhub_name: str,
+        shared_access_signature: str,
+        shared_access_key_name: str,
+        shared_access_key: str,
```

Comment:
apologies on this one, checked with Swathi in the other PR and there can be some None parts in a connection string. 

```suggestion
        eventhub_name: Optional[str] = None,
        shared_access_signature: Optional[str] = None,
        shared_access_key_name: Optional[str] = None,
        shared_access_key: Optional[str] = None,
```